src/Cli.Tests/EndToEndTests.cs

```diff
@@ -296,6 +296,34 @@ public void TestAddTelemetry(string? appInsightsEnabled, string appInsightsConnS
         Assert.AreEqual("InstrumentationKey=00000000", updatedConfig.Runtime.Telemetry.ApplicationInsights.ConnectionString);
     }
 
+    /// <summary>
+    /// Test to verify sucess or failure based on different values for
+    /// depth limit in GraphQL runtime settings.
+    /// Valid values are [1, INT32.MAX_VALUE], and -1 to remove depth limit.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow("8", true, DisplayName = "Succesful update with a valid value for depth limit")]
+    [DataRow("0", false, DisplayName = "Failure as depth limit cannot be set to 0.")]
+    [DataRow("-1", true, DisplayName = "Succesful update to to remove depth limit using -1.")]
+    [DataRow("-15", false, DisplayName = "Failure as negative value other than -1 is invalid")]
+    [DataRow("2147483647", true, DisplayName = "Succesful update setting value to INT32_MAX")]
+    [DataRow("2147483648", false, DisplayName = "Failure when using depth value greater than INT32_MAX")]
+    [DataRow("seven", false, DisplayName = "Failure when using string value for depth limit")]
+    public void TestUpdateDepthLimitInGraphQLRuntimeSettings(string depthLimit, bool isSuccess)
+    {
+        string[] initArgs = { "init", "-c", TEST_RUNTIME_CONFIG_FILE, "--host-mode", "development", "--database-type",
+            "mssql", "--connection-string", TEST_ENV_CONN_STRING };
+        Program.Execute(initArgs, _cliLogger!, _fileSystem!, _runtimeConfigLoader!);
+
+        Assert.IsTrue(_runtimeConfigLoader!.TryLoadConfig(TEST_RUNTIME_CONFIG_FILE, out RuntimeConfig? runtimeConfig));
+        Assert.IsNotNull(runtimeConfig);
+
+        string[] runtimeArgs = { "runtime", "-c", TEST_RUNTIME_CONFIG_FILE, "--depth-limit", depthLimit };
+        int isError = Program.Execute(runtimeArgs, _cliLogger!, _fileSystem!, _runtimeConfigLoader!);
```

Comment:
probably worth denoting that this line is your actual test: the `// Act` section.